pkg/users/accounts-list.js

```diff
@@ -183,17 +236,82 @@
+const GroupsList = ({ groups, accounts }) => {
+    const [isExpanded, setIsExpanded] = useState(false);
+    const columns = [
+        { title: _("Group name"), sortable: true },
+        { title: _("ID"), sortable: true },
+        { title: _("# of users"), sortable: true },
+        { title: _("Accounts") },
+    ];
 
-    const accounts = mapGroupsToAccount(accountsInfo, groups);
+    const sortRows = (rows, direction, idx) => {
+        // GID and members columns are numeric
+        const isNumeric = idx == 1 || idx == 2;
+        const sortedRows = rows.sort((a, b) => {
+            const aitem = a.columns[idx].sortKey || a.columns[idx].title;
+            const bitem = b.columns[idx].sortKey || b.columns[idx].title;
+            const aname = a.columns[0].sortKey;
+            const bname = b.columns[0].sortKey;
 
-    const filtered_accounts = accounts.filter(account => {
-        if ((account.uid < 1000 && account.uid !== 0) ||
-                 account.shell.match(/^(\/usr)?\/sbin\/nologin/) ||
-                 account.shell === '/bin/false')
-            return false;
+            // administrator groups are always first
+            if (admins.includes(aname))
+                return direction === SortByDirection.asc ? -1 : 1;
+            if (admins.includes(bname))
+                return direction === SortByDirection.asc ? 1 : -1;
 
+            if (isNumeric)
+                return bitem - aitem;
+            else
+                return aitem.localeCompare(bitem);
+        });
+        return direction === SortByDirection.asc ? sortedRows : sortedRows.reverse();
+    };
+
+    return (
+        <Card className="ct-card" isExpanded={isExpanded}>
+            <CardHeader
+                className="ct-card-expandable-header"
+                onExpand={() => setIsExpanded(!isExpanded)}
+                toggleButtonProps={{
+                    id: 'groups-view-toggle',
+                    'aria-label': _("Groups"),
+                    'aria-expanded': isExpanded
+                }}>
+                <CardTitle className="pf-l-flex pf-m-space-items-sm pf-m-align-items-center">
+                    <Text component={TextVariants.h2}>{_("Groups")}</Text>
+                    {!isExpanded && <>
+                        {groups.slice(0, 3)
+                                .map(group => {
+                                    const color = group.isAdmin ? "gold" : "cyan";
+                                    return (
+                                        <Label key={group.name} variant="filled" color={color}>
+                                            {group.name + ": " + (group.userlistPrimary.length + group.userlist.length)}
+                                        </Label>
+                                    );
+                                })}
+                        <Button key="more" className="group-more-btn" isInline variant='link' onClick={() => setIsExpanded(!isExpanded)}>
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17996-20221206-122041-ea567d53-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)